src/client.rs

```diff
@@ -246,14 +263,23 @@ pub async fn run(config: ClientConfig) -> Result<()> {
         config: Arc::new(config),
     });
 
-    let client = Client {
-        plugin_senders: Mutex::new(plugin_senders),
-        state: Mutex::new(state),
+    let client = Arc::new(Client {
+        state: Mutex::new(state.clone()),
         db_client: Arc::new(db_client),
-    };
+        sender: tx_sender.clone(),
+    });
+
+    let (plugin_senders, plugin_tasks): (Vec<_>, Vec<_>) =
+        crate::plugin::load(client.clone(), &state.config)?
+            .into_iter()
+            .unzip();
+
+    // Now that we've used the config to load the plugins, we want to explicitly
+    // drop our reference to the state.
+    drop(state);
```

Comment:
Github is being very annoying today, so I can actually suggest on the main lines

I think that typically rust is pretty good about dropping things that are unused
https://dtolnay.github.io/rust-quiz/12
https://dtolnay.github.io/rust-quiz/19
but I think in this case it may not be dropped until the end of the method

You can use blocks for this if you want, something like this
```
let (plugin_senders, plugin_tasks): ... = {
   let state = ...;
   let loaded = plugin::load(...);
   loaded
};
```
You'll have to deal with returning the client as well, which is a bit annoying, but you can just make a 3-tuple
It IS an arc tho, so drop may be the cleanest answer

But, why bother dropping it? the clone of the arc is going to exist in the plugin tasks usually forever right?

Reply:
Isn't there also a clone of the state in the client that will be dropped at the same time as the original state var?
I feel like rust makes dangling references something to not worry about

Reply:
Deleted my comment as @guswynn already covered it. @belak to @guswynn's point, is this necessary to fix forward now? I'd imagine things would be clearer in the future if and when dynamic config updates ever get updated, instead of trying to plan for eventualities right now.